msgpack.js

```diff
@@ -547,8 +547,12 @@
 		// Node.js
 		module.exports = msgpack;
 	}
-	else {
-		// Global object
+	else if (typeof WorkerGlobalScope === "function" && self instanceof WorkerGlobalScope) {
+		// Web Worker scope
+		self.msgpack = msgpack;

```

Comment:
Would it make sense (or is it even possible) to let the name "msgpack" be configured in this case as well like for the browser below?

Reply:
No, since a web worker runs in a different execution thread and has no access to the `window` object.

Reply:
@ygoe any news on the PR?

> even on main document, the `self` or `globalThis` is valid, can check and use it in both world?

Not sure what do you mean.
I am using the `self` object because of how script importing works in a Web Worker scope.

In a browser scope, I am not sure what would the difference be between using `self` or `window`, but I'm guessing it's the same.

Reply:
Looks like `self` [has been here since forever](https://developer.mozilla.org/en-US/docs/Web/API/Window/self). Didn't know that.

So maybe the solution would be to just replace `window` with `self` everywhere? No additional check case needed.